lib/Doctrine/Migrations/Tools/TransactionHelper.php

```diff
@@ -57,10 +59,13 @@ public static function rollbackIfInTransaction(Connection $connection): void
 
     private static function inTransaction(Connection $connection): bool
     {
-        $wrappedConnection = $connection->getWrappedConnection();
+        $innermostConnection = $connection;
+        while (method_exists($innermostConnection, 'getWrappedConnection')) {
+            $innermostConnection = $innermostConnection->getWrappedConnection();
+        }
 
         /* Attempt to commit or rollback while no transaction is running
            results in an exception since PHP 8 + pdo_mysql combination */
-        return ! $wrappedConnection instanceof PDO || $wrappedConnection->inTransaction();
+        return ! $innermostConnection instanceof PDO || $innermostConnection->inTransaction();
```

Comment:
If the "in transaction" method for the drivers of the platform that don't support transactional DDL is still of interest, it looks like the right course of action is to revive the [discussion](https://externals.io/message/114181) in PHP internals.

Reply:
We are planning to drop the `TransactionHelper` class in the next major release ([ref](https://github.com/doctrine/migrations/pull/1170)), so I don't think it will be of interest.